node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/util/HttpException.java

```diff
@@ -12,25 +13,33 @@ public NotFoundException(String message) {
         }
     }
 
-    static void throwOnFailure(int statusCode, String message) {
+    /**
+     * Returns empty on success.
+     * Returns an exception if the error is retriable.
+     * Throws an exception on a non-retriable error, like 404 Not Found.
+     */
+    static Optional<HttpException> handleStatusCode(int statusCode, String message) {
```

Comment:
Could do that. I mostly wanted to avoid another try-catch in this function given its already littered with try-catch/finally, but also to avoid the overhead of throws (stack generation). Perhaps that gets optimized away.